src/suricata.c

```diff
@@ -1111,20 +1148,30 @@ static void SCPrintElapsedTime(SCInstance *suri)
 static int ParseCommandLineAfpacket(SCInstance *suri, const char *optarg)
 {
 #ifdef HAVE_AF_PACKET
-    if (suri->run_mode == RUNMODE_UNKNOWN) {
-        suri->run_mode = RUNMODE_AFP_DEV;
+    RunModesList *runmodes = &suri->runmodeslist;
+    if (runmodes->run_mode[runmodes->runmodes_cnt] == RUNMODE_UNKNOWN) {
+        if (!RunmodeIsAlreadyRan(runmodes, RUNMODE_AFP_DEV)) {
```

Comment:
IsAlreadyRan is a bit of a confusing name, what does 'Ran' mean here? What about ..AlreadyAdded? of ..IsSet or something similar.